packages/prediction-app/src/components/ChapValidator/ChapValidator.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useEffect } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import { Button, CircularLoader, IconError24 } from '@dhis2/ui'
+import { useNavigate } from 'react-router-dom'
+import styles from './ChapValidator.module.css'
+import { useChapStatus } from '../../features/settings/ChapSettings/hooks/useChapStatus'
+import { useRoute } from '../../hooks/useRoute'
+
+type Props = {
+    children: React.ReactNode
+}
+
+export const ChapValidator = ({ children }: Props) => {
+    const { route } = useRoute()
+    const chapStatus = route ? useChapStatus({ route }) : null
+    const status = chapStatus?.status
+    const error = chapStatus?.error
+    const isLoading = chapStatus?.isLoading || false
```

Comment:
It seems like the reason you added all of this is due to route possibly being undefined. Instead of doing all this, please revert to `const { status, error, isLoading } = useChapStatus({ route })`. This will give you a type error saying that `Type 'Route | undefined' is not assignable to type 'Route'`. Please fix this type error by changing the hook so that it can take in route or undefined. However, please also add the `enabled: !!route` flag on the `useQuery` hook, so that it is only triggered once the Route is loaded.